app/services/workflow_executions/preparation_service.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+require 'csv'
+require 'tempfile'
+
+module WorkflowExecutions
```

Comment:
Can you add some basic error checking for now such as the samplesheet isn't empty, directory was created successfully, attachments copied successfully etc

Reply:
Not sure what you are asking for? If adding it to the actual service this would be a pain as I would have to use the `ActiveStorage::Service` directly and handle it differently for each service. Or are you suggesting for the test?